core/src/main/java/com/amplifyframework/api/graphql/OperationType.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.graphql;
+
+/**
+ * Represents a GraphQL operation type.
+ */
+public interface OperationType {
+    /**
+     * Returns the name of the operation to be used in the GraphQL document (i.e. query, subscription, or mutation).
```

Comment:
I like the idea to create an enum for the operations (query, mutation, subscription), so that the possible operations are constrained, so I've made that change.

As for the second part of this suggestion, I do see the benefit of guaranteeing that `Operation` is only implemented by an enum of a type constrained to a `OperationType`.   However, I think it's overly complicated for now, since we'd have to introduce a new set of objects (Query, Mutation, Subscription), that wrap the type enums (QueryType, MutationType, SubscriptionType).  Not opposed to exploring again in the future though.